src/rhsm/config.py

```diff
@@ -107,54 +107,26 @@
 
 def in_container() -> bool:
     """
-    Are we running in a docker container or not?
+    Are we running in a container or not?
     """
-    # For development in containers we must be able to turn container detection
-    # off
+    # For development in containers we must be able to turn container detection off
     if os.environ.get("SMDEV_CONTAINER_OFF", ""):
         return False
 
-    def in_ocp() -> bool:
-        """
-        Is the system running as pod in OCP (OpenShift Container Platform)?
-
-        Check some of the canonical environment variables set by Kubernets
-        (on which OCP is based):
-        https://kubernetes.io/docs/concepts/containers/container-environment/
-        in particular, look for the "kubernetes" default service
+    # If the path exists, we are in a container.
+    # In UBI containers (RHEL, CentOS), path HOST_CONFIG_DIR='/etc/rhsm-host/'
+    # is a symlink to /run/secrets/rhsm. That path is a symlink/Podman secret
+    # specified in /usr/share/containers/mounts.conf, pointing to host's directory
+    # /usr/share/rhel/secrets. The directories inside are themselves symlinks
+    # to other host directories populated by subscription-manager.
+    # If this secret (= the container directory /etc/rhsm-host/) exists,
+    #   the system is considered to be a container.
+    # If this secret does not exist,
+    #   the system is considered to be a non-container.
+    if os.path.isdir(HOST_CONFIG_DIR):
+        log.debug(f"Secret {HOST_CONFIG_DIR} exists, container detected.")
```

Comment:
"secret <path> exists" feels like it's something shady :)) maybe "found <path> with certificates"?